### ISSUE TITLE
vue/html-quotes does not accept Number values

### ISSUE THREAD

**Issue by mtwolak** [vuejs/eslint-plugin-vue#1098]
**Tell us about your environment**	
* **ESLint version:**	6.8.0
* **eslint-plugin-vue version:**	6.2.2
* **Node version:**	10.17.0

**Please show your full configuration:**
```json
module.exports = {
    root: true,
    env: {node: true,
      browser: true
    },
    parserOptions: {
      parser: "babel-eslint",
      sourceType: "module"
    },
    extends: [
      "eslint:recommended",
      "plugin:vue/recommended"
    ],
  };
```

**What did you do?**	
[Full code on github](https://github.com/mtwolak/eslint-html-quotes)
Parent component:
```vue
<template>
  <div id="app">
    <CustomComponent num=10 />
  </div>
</template>

<script>

import CustomComponent from './components/CustomComponent'

export default {
  name: 'App',
  components: {
    CustomComponent
  }
}
</script>

```

Child component:
```vue
<template>
  <div>{{ num }}</div>
</template>

<script>
export default {
  name: 'CustomComponent',
  props: {
    num: {
      type: Number,
      required: true,
      default: 5
    }
  }
}
</script>

```

**What did you expect to happen?**
I cannot pass 10 as Number to child component with enabled rule `vue/html-quotes`. It complains it should be String.
 

**What actually happened?**
- If I write
```vue
<CustomComponent num=10 />
```
it complains 10 is not String: `Expected to be enclosed by double quotes`

- If I write 
```vue
<CustomComponent num="10" />
```
I got an error 
```
[Vue warn]: Invalid prop: type check failed for prop "num". Expected Number with value 10, got String with value "10".
found in
---> <CustomComponent> at src/components/CustomComponent.vue
       <App> at src/App.vue
         <Root>
```

**Comment by ota-meshi** [comment]
This issue is not related to the eslint-plugin-vue report.

You need to implement it like this:

```vue
<CustomComponent :num="10" />
```

It will compile to a string whether or not the attribute has quotes.

https://template-explorer.vuejs.org/#%3CCustomComponent%20num%3D10%20%2F%3E

You can compile to a number using the v-bind directive.

https://template-explorer.vuejs.org/#%3CCustomComponent%20%3Anum%3D10%20%2F%3E
